pkgs/applications/misc/blender/derivation.nix

```diff
@@ -0,0 +1,268 @@
+{ config, stdenv, lib, fetchFromGitea, fetchzip, boost, cmake, ffmpeg, gettext, glew
+, ilmbase, libepoxy, libXi, libX11, libXext, libXrender
+, libjpeg, libpng, libsamplerate, libsndfile
+, libtiff, libwebp, libGLU, libGL, openal, opencolorio, openexr, openimagedenoise, openimageio, openjpeg, python310Packages
+, openvdb, libXxf86vm, tbb, alembic
+, zlib, zstd, fftw, opensubdiv, freetype, jemalloc, ocl-icd, addOpenGLRunpath
+, jackaudioSupport, libjack2
+, cudaSupport, cudaPackages ? { }
+, hipSupport, hip # comes with a significantly larger closure size
+, colladaSupport, opencollada
+, spaceNavSupport, libspnav
+, makeWrapper
+, pugixml, llvmPackages, SDL, Cocoa, CoreGraphics, ForceFeedback, OpenAL, OpenGL
+, waylandSupport, pkg-config, wayland, wayland-protocols, libffi, libdecor, libxkbcommon, dbus
+, potrace
+, openxr-loader
+, embree, gmp, libharu
+, openpgl
+, mesa
+, runCommand
+, version, hashes, isLTS
+}:
+
+let
+  python = python310Packages.python;
```

Comment:
I think its better to parametrize this pythonPackages thing and then expose it in passthrough.

All blender 3.x so far are based on python 3.10 but it may be not true in the near future then it would be a tech debt for the next touching this and a wrapper generator might get things wrong.

Reply:
Good idea.

Reply:
.... I've realized that I have no idea how to test `withPackages`, so I need your help. I have no idea if it works.

I also don't know how to help you modify your package so it works with the new system.

Reply:
The same way you would test `python3.withPackages`, the only difference is that the blender derivation hints about which python version it's based and pass the package scope to the function you pass to withPackages. That's the reason it exists.

And to open a python REPL inside blender you run it with `blender --background --python-console`. You can also pass a script with `blender --python script.py` to check for the imports.